projectgenerator/libqgsprojectgen/generator/postgres/postgrescreator.py

```diff
@@ -187,6 +188,17 @@ def layers(self):
                     field.widget = 'TextEdit'
                     field.widget_config['IsMultiline'] = True
 
+                if 'time' in fielddef['data_type'] or 'date' in fielddef['data_type']:
+                    field.widget = 'DateTime'
+                    field.widget_config['calendar_popup'] = True
+                    # timestamps don't need further configuration
+                    if 'time' in fielddef['data_type'] and not 'timestamp' in fielddef['data_type']:
+                        field.widget_config['field_format'] = 'HH:mm:ss'
+                        field.widget_config['display_format'] = 'HH:mm:ss'
+                    elif 'date' in fielddef['data_type']:
+                        field.widget_config['field_format'] = 'yyyy-MM-dd'
+                        field.widget_config['display_format'] = 'yyyy-MM-dd'
```

Comment:
Actually, I'm also one of those edge cases...

Apparently it's only the default `en(_US)`

```py
>>> QLocale('en').dateFormat(QLocale.ShortFormat)
'M/d/yy'
>>> QLocale('en_GB').dateFormat(QLocale.ShortFormat)
'dd/MM/yyyy'
```

Let's do it anyway and blame it on Qt :P